plugins/processors/awsappsignals/internal/resolver/ec2_test.go

```diff
@@ -0,0 +1,148 @@
+	envAttr, ok := attributes.Get(attr.HostedInEC2Environment)
+	assert.True(t, ok)
+	assert.Equal(t, asgName, envAttr.AsString())
+}
+
+func TestEC2HostedInAttributeResolverWithNoConfiguredName_NoASG_EnvExists(t *testing.T) {
+	resolver := newEC2HostedInAttributeResolver("")
+
+	envName := "my-env"
+	attributes := pcommon.NewMap()
+	resourceAttributes := pcommon.NewMap()
+	resourceAttributes.PutStr(attr.AWSHostedInEnvironment, envName)
+
+	resolver.Process(attributes, resourceAttributes)
+	envAttr, ok := attributes.Get(attr.HostedInEC2Environment)
+	assert.True(t, ok)
+	assert.Equal(t, envName, envAttr.AsString())
+}
+
+func TestEC2HostedInAttributeResolverWithConfiguredName_NoASG_EnvExists(t *testing.T) {
+	resolver := newEC2HostedInAttributeResolver("test")
+
+	envName := "my-env"
+	attributes := pcommon.NewMap()
+	resourceAttributes := pcommon.NewMap()
+	resourceAttributes.PutStr(attr.AWSHostedInEnvironment, envName)
+
+	resolver.Process(attributes, resourceAttributes)
+	envAttr, ok := attributes.Get(attr.HostedInEC2Environment)
+	assert.True(t, ok)
+	assert.Equal(t, envName, envAttr.AsString())
+}
+
+func TestEC2HostedInAttributeResolverWithNoConfiguredName_ASGExists_EnvExists(t *testing.T) {
+	resolver := newEC2HostedInAttributeResolver("")
+
+	asgName := "ASG"
+	envName := "my-env"
+	attributes := pcommon.NewMap()
+	resourceAttributes := pcommon.NewMap()
+	resourceAttributes.PutStr(attr.EC2AutoScalingGroupName, asgName)
+	resourceAttributes.PutStr(attr.AWSHostedInEnvironment, envName)
+
+	resolver.Process(attributes, resourceAttributes)
+	envAttr, ok := attributes.Get(attr.HostedInEC2Environment)
+	assert.True(t, ok)
+	assert.Equal(t, envName, envAttr.AsString())
+}
+
+func TestEC2HostedInAttributeResolverWithConfiguredName_ASGExists_EnvExists(t *testing.T) {
+	resolver := newEC2HostedInAttributeResolver("test")
+
+	asgName := "ASG"
+	envName := "my-env"
+	attributes := pcommon.NewMap()
+	resourceAttributes := pcommon.NewMap()
+	resourceAttributes.PutStr(attr.EC2AutoScalingGroupName, asgName)
+	resourceAttributes.PutStr(attr.AWSHostedInEnvironment, envName)
+
+	resolver.Process(attributes, resourceAttributes)
+	envAttr, ok := attributes.Get(attr.HostedInEC2Environment)
+	assert.True(t, ok)
+	assert.Equal(t, envName, envAttr.AsString())
+}
+
+func TestEC2HostedInAttributeResolverWithResourceDetectionAttributes(t *testing.T) {
+	resolver := newEC2HostedInAttributeResolver("")
+
+	attributes := pcommon.NewMap()
+	resourceAttributes := pcommon.NewMap()
+	resourceAttributes.PutStr(attr.ResourceDetectionHostId, "hostid")
+	resourceAttributes.PutStr(attr.ResourceDetectionHostName, "hostname")
+	resourceAttributes.PutStr(attr.ResourceDetectionASG, "asg")
+
+	resolver.Process(attributes, resourceAttributes)
+	expectedInstanceId, ok := attributes.Get(attr.EC2InstanceId)
+	assert.True(t, ok)
+	assert.Equal(t, "hostid", expectedInstanceId.AsString())
+
+	expectedHostName, ok := attributes.Get(attr.ResourceDetectionHostName)
```

Comment:
added this unit test from feedback, if I removed the resource detection host name from the resolver map, then this fails